pandas/tests/arrays/test_boolean.py

```diff
@@ -130,8 +130,8 @@ def test_to_boolean_array_missing_indicators(a, b):
     ],
 )
 def test_to_boolean_array_error(values):
-    # error in converting existing arrays to BooleanArray
-    with pytest.raises(TypeError):
+    msg = "Error converting existing arrays to BooleanArray"
+    with pytest.raises(TypeError, match=msg):
```

Comment:
Thanks @amy-graham-js , and welcome :)

The comment
```
# error in converting existing arrays to BooleanArray
```
looks like more of a comment on the source than the expected error, so we should keep it as a comment

To find out which error is actually raised, you could try inserting a [breakpoint](https://docs.python.org/3/library/pdb.html):
```
def test_to_boolean_array_error(values):
    # error in converting existing arrays to BooleanArray
    breakpoint()
    with pytest.raises(TypeError):
        pd.array(values, dtype="boolean")
```
, running the test:
```
pytest pandas/tests/arrays/test_boolean.py
```
and then, when the breakpoint is reached, executing the command
```
pd.array(values, dtype="boolean")
```
to see what error is raised

Reply:
Thanks @MarcoGorelli for your help! Hopefully that should be ok now :)